From: Charles <charles05@canonical.com>
Date: Thu, 16 Jul 2026 22:26:04 +0100
Subject: credentials/file: Detect changes at runtime

Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+bug/2156893
Bug: https://gitlab.gnome.org/GNOME/gnome-remote-desktop/-/work_items/279
Forwarded: https://gitlab.gnome.org/GNOME/gnome-remote-desktop/-/merge_requests/407
---
 src/grd-credentials-file.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/grd-credentials-file.c b/src/grd-credentials-file.c
index 0093239..d6673ed 100644
--- a/src/grd-credentials-file.c
+++ b/src/grd-credentials-file.c
@@ -91,6 +91,13 @@ grd_credentials_file_lookup (GrdCredentials      *credentials,
   g_autofree const char *serialized = NULL;
   g_autoptr (GError) local_error = NULL;
 
+  /* Reload from disk to pick up changes made by other processes. */
+  g_key_file_load_from_file (credentials_file->key_file,
+                             credentials_file->filename,
+                             G_KEY_FILE_KEEP_COMMENTS |
+                             G_KEY_FILE_KEEP_TRANSLATIONS,
+                             NULL);
+
   serialized = g_key_file_get_string (credentials_file->key_file,
                                       group_name_from_type (type),
                                       GRD_CREDENTIALS_FILE_KEY, &local_error);
